src/codebashing/codebashing.ts

```diff
@@ -0,0 +1,54 @@
+import * as vscode from "vscode";
+import { getCodeBashing } from "../ast/ast";
+import { Logs } from "../models/logs";
+import {
+  constants
+} from "../utils/common/constants";
+import AstError from "../exceptions/AstError";
+
+const CODEBASHING_NO_LICENSE =
+  "You don't have a license for Codebashing. Please Contact your Admin for the full version implementation. Meanwhile, you can use the link below.";
+const CODEBASHING_NO_LESSON = "Currently, this vulnerability has no lesson";
+
+const PROGRESS_HEADER: vscode.ProgressOptions = {
+  location: vscode.ProgressLocation.Notification,
+  title: "Getting codebashing link",
+  cancellable: false,
+};
+
+export async function getCodebashingLink(
+  cweId: string,
+  language: string,
+  queryName: string,
+  logs: Logs
+) {
+  vscode.window.withProgress(PROGRESS_HEADER, async (progress, token) => {
+    token.onCancellationRequested(() => logs.info("Canceled loading"));
+    try {
+      logs.info("Fetching codebashing link");
+      const codeBashingArray = await getCodeBashing(cweId, language, queryName);
+      vscode.env.openExternal(vscode.Uri.parse(codeBashingArray?.path));
+    } catch (err) {
+      logs.error("Failed getting codebashing link");
+      if (err instanceof AstError) {
+        if (err.code === constants.astErrorCodeBashingNoLicense) {
+          vscode.window
+            .showInformationMessage(CODEBASHING_NO_LICENSE, "Codebashing")
+            .then(() =>
+              vscode.env.openExternal(
+                vscode.Uri.parse("https://free.codebashing.com")
+              )
+            );
+          return;
+        } else if (err.code === constants.astErrorCodeBashingNoLesson) {
+          vscode.window.showInformationMessage(CODEBASHING_NO_LESSON);
+          return;
+        }
+      }
+
+      vscode.window.showWarningMessage(
+        String(err).replace(constants.errorRegex, "").replaceAll("\n", "")
```

Comment:
@tiagobcx  maybe this could be a util ?

Reply:
It is not used elsewhere, should still go to the utils?